core/ecschema-metadata/src/Context.ts

```diff
@@ -489,4 +525,73 @@ export class SchemaContext implements ISchemaItemLocater {
   public getKnownSchemas(): Schema[] {
     return this._knownSchemas.getAllSchemas();
   }
+
+  /**
+   * Get metadata for a schema item. This method will load the metadata from the schema context if necessary.
+   *
+   * @param classFullName The full name of the class in the format "SchemaName:SchemaItemName".
+   * @param itemConstructor The constructor of the schema item to return.
+   * @returns The requested schema item metadata.
+   * @throws [[ECObjectsError]] if the metadata cannot be found or loaded.
+   *
+   * @example
+   * ```typescript
+   * const entityClass = schemaContext.getSchemaItemMetaData("BisCore:Element", EntityClass);
+   * const customAttributeClass = schemaContext.getSchemaItemMetaData("BisCore:SchemaHasBehavior", CustomAttributeClass);
+   * const relationshipClass = schemaContext.getSchemaItemMetaData("BisCore:ElementOwnsChildElements", RelationshipClass);
+   * ```
+   */
+  public getSchemaItemMetaData<T extends typeof SchemaItem>(classFullName: string, itemConstructor: T): InstanceType<T> {

```

Comment:
Hmm, I do not like this one. It feels like clutter.
I'd rather not have a fullname overload and instead have a nice public method which splits a fullname into two strings.
So instead of
`x.getSchemaItemMetaData("myschema:myclass");`
people could call
`x.getItem(...parseFullName("mySchema:myClass");`

Assuming the parseFullName method looks like this:
```ts
  public parseFullName(input: string): [string, string] {
    const [first, second] = input.split(":");
    return [first, second];
  }
The parseFullName is just a sketch. The real one needs to support colon, dot and make sure there are only 2 results...
```
Or if we absolutely need to stick with this, better call it getItemByFullName(). Everything here is MetaData, so that previous name is ambiguous.